docs/ref/models/lookups.txt

```diff
@@ -44,30 +44,48 @@ register lookups on itself. The two prominent examples are
-        ``DateField.register_lookup(YearExact)`` will register ``YearExact``
-        lookup on ``DateField``. It overrides a lookup that already exists with
-        the same name. ``lookup_name`` will be used for this lookup if
+        Registers a new lookup in the class or a class instance. For example,
+        registering a lookup to a class would be done by::
+
+        ``DateField.register_lookup(YearExact)``
+
+        This will register ``YearExact`` lookup on ``DateField``. It overrides a
+        lookup that already exists in the class with the same ``lookup_name``.
+
+        To register a lookup to an instance of a class::
+
+        ``datetime_field_instance.register_lookup(MonthExact)``
+
+        This will register ``MonthExact`` lookup on ``Model.date``, where
+        ``datetime_field_instance`` is a an instance of ``DateField``. This would
+        make the ``MonthExact`` lookup available only on this instance of
+        ``DateField``. A Lookup registered on an instance would override its
+        an already existing "instance" lookup or "class" lookup that already has
+        the same ``lookup_name``. The field instance of a Model can be accessed
+        by the :ref:`Field Access API <model-meta-field-api>`.
+
+        In both the cases, ``lookup_name`` will be used for this lookup if
         provided, otherwise ``lookup.lookup_name`` will be used.
 
     .. method:: get_lookup(lookup_name)
 
-        Returns the :class:`Lookup` named ``lookup_name`` registered in the class.
-        The default implementation looks recursively on all parent classes
-        and checks if any has a registered lookup named ``lookup_name``, returning
-        the first match.
+        Returns the :class:`Lookup` named ``lookup_name`` registered in the class
+        or a class instance depending on what calls it. The default implementation
+        looks recursively on all parent classes and checks if any has a
+        registered lookup named ``lookup_name``, returning the first match. Instance
+        lookups would override any class lookups with the same ``lookup_name``.
 
     .. method:: get_lookups()
 
-        Returns a dictionary of each lookup name registered in the class mapped
-        to the :class:`Lookup` class.
+        Returns a dictionary of each lookup name registered in the class or a
+        class instance mapped to the :class:`Lookup` class.
 
     .. method:: get_transform(transform_name)
 
-        Returns a :class:`Transform` named ``transform_name``. The default
-        implementation looks recursively on all parent classes to check if any
-        has the registered transform named ``transform_name``, returning the first
-        match.
+        Returns a :class:`Transform` named ``transform_name`` registered in the
+        class. The default implementation looks recursively on all parent classes
+        to check if any has the registered transform named ``transform_name``,
+        returning the first match.
```

Comment:
`versionchanged` annotations are missing. We should add them to changed methods or an overall annotation in the `Registration API`, e.g.

```
.. versionchanged:: 4.2

    Support for registering lookups on instances was added.
```